### PR TITLE
feat: add guild directory support

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

Adds `GUILD_DIRECTORY` channel type, and the `HUB` guild feature.

- https://github.com/discord/discord-api-docs/pull/4701

**Status and versioning classification:**
Please move lines that apply to you out of the comment:
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Comment by iCrawl** [comment]
Is there no upstream PR for this?

**Comment by suneettipirneni** [comment]
Upstream PR merged

**Review by SpaceEEC — APPROVED** [review]

File: packages/discord.js/src/structures/Channel.js [hunk 1]
```diff
@@ -260,6 +262,10 @@ class Channel extends Base {
             if (!allowUnknownGuild) channel.parent?.threads.cache.set(channel.id, channel);
             break;
           }
+          // TODO: Use discord-api-types
+          case 14:
```
Comment (kyranet) [inline packages/discord.js/src/structures/Channel.js:266]: This should be available in next release, as https://github.com/discordjs/discord-api-types/pull/399 includes it. 👀

File: packages/discord.js/src/structures/DirectoryChannel.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+const { Channel } = require('./Channel');
+
+/**
+ * Represent a channel that displays a directory of guilds
+ */
+class DirectoryChannel extends Channel {
```
Comment (kyranet) [inline packages/discord.js/src/structures/DirectoryChannel.js:8]: Aren't directory channels, guild channels?

Comment (Jiralite) [reply packages/discord.js/src/structures/DirectoryChannel.js:8]: I don't believe so. Additionally, we wouldn't be able to use any of the methods a guild channel offers as the only way to receive this is to fetch a hub invite.

Comment (suneettipirneni) [reply packages/discord.js/src/structures/DirectoryChannel.js:8]: Basically what Jira said

**Review by Jiralite — COMMENTED** [review]
Just remembered something. Let's add a `Channel#isDirectory()` method just like all the other channel type checks!

File: packages/discord.js/src/structures/Channel.js (hunk 1 above)
Comment (vladfrangu) [reply packages/discord.js/src/structures/Channel.js:275]: Released

File: packages/discord.js/src/structures/Channel.js [hunk 3]
```diff
@@ -173,6 +174,15 @@ class Channel extends Base {
     return this.type === ChannelType.GuildStageVoice;
   }
 
+  /**
+   * Indicates whether this channel is a {@link DirectoryChannel}
+   * @returns {boolean}
+   */
+  isDirectory() {
+    // TODO: Use dapi-types
+    return this.type === 14;
```
Comment (imranbarbhuiya) [inline packages/discord.js/src/structures/Channel.js:183]:
```suggestion
    return this.type === ChannelType.GuildDirectory;
```

**Review by imranbarbhuiya — CHANGES_REQUESTED** [review]

**Review by Jiralite — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**